src/stores/search.ts

```diff
@@ -0,0 +1,341 @@
+      // Ensure that q filter always comes first
+      { q: state.searchTerm.trim() } as ApiQueryParams
+    )
+  })
+
+  const allFilterCategories = Object.keys(state.filters) as FilterCategory[]
+  /**
+   * Initial filters do not include the provider filters. We create the provider filters object
+   * when we fetch the provider data on the Nuxt server initialization.
+   * We call this function to reset the filters to the initial base filters AND the provider filters.
+   */
+  const getBaseFiltersWithProviders = (): Filters => {
+    const resetProviders = (mediaType: SupportedMediaType): FilterItem[] => {
+      return state.filters[`${mediaType}Providers`].map((provider) => ({
+        ...provider,
+        checked: false,
+      }))
+    }
+    return {
+      ...clonedeep(filterData),
+      audioProviders: resetProviders(AUDIO),
+      imageProviders: resetProviders(IMAGE),
+    }
+  }
+
+  /**
+   * Returns the number of checked filters, excluding the `mature` filter.
+   */
+  const appliedFilterCount: ComputedRef<number> = computed(() => {
+    const filterKeys = mediaFilterKeys[state.searchType].filter(
+      (f) => f !== 'mature'
+    )
+    return filterKeys.reduce((count, filterCategory) => {
+      return (
+        count + state.filters[filterCategory].filter((f) => f.checked).length
+      )
+    }, 0)
+  })
+
+  /**
+   * True if any filter for selected search type except `mature` is checked.
+   */
+  const isAnyFilterApplied: ComputedRef<boolean> = computed(() => {
+    const filterEntries = Object.entries(searchFilters.value) as [
+      string,
+      FilterItem[]
+    ][]
+    return filterEntries.some(
+      ([filterKey, filterItems]) =>
+        filterKey !== 'mature' && filterItems.some((filter) => filter.checked)
+    )
+  })
+
+  // Actions
+  /**
+   * After a search type is changed, unchecks all the filters that are not
+   * applicable for this Media type.
+   */
+  function clearOtherMediaTypeFilters(searchType: SupportedSearchType) {
+    const mediaTypesToClear = supportedSearchTypes.filter(
+      (type) => type !== searchType
+    )
+    const filterKeysToClear = mediaTypesToClear.reduce((acc, type) => {
+      return [...acc, ...mediaUniqueFilterKeys[type]]
+    }, [] as FilterCategory[])
+
+    allFilterCategories.forEach((filterCategory) => {
+      if (filterKeysToClear.includes(filterCategory)) {
+        state.filters[filterCategory] = state.filters[filterCategory].map(
+          (f) => ({
+            ...f,
+            checked: false,
+          })
+        )
+      }
+    })
+  }
+
+  /**
+   * Replaces filters with the newFilterData parameter, making sure that
+   * audio/image provider filters are handled correctly.
+   */
+  function replaceFilters(newFilterData: Filters) {
+    const providerFilters: FilterCategory[] = [
+      'audioProviders',
+      'imageProviders',
+    ]
+
+    allFilterCategories.forEach((filterCategory) => {
+      if (providerFilters.includes(filterCategory)) {
```

Comment:
When would this be false :thinking:

Reply:
This was originally written to prevent the provider codes that do not exist in the filters for current search type from breaking this function. For example, going to `/search/audio?source=imagesource` would cause a problem because `imagesource` existed in the `imageProviders` but did not exist in `audioProviders`.
I've improved the `queryToFilterData` to prevent that, and added more documentation there in the last commit.